datafusion-examples/examples/query_planning/expr_api.rs

```diff
@@ -551,7 +554,9 @@ fn type_coercion_demo() -> Result<()> {
     assert!(physical_expr.evaluate(&batch).is_ok());
 
     // 2. Type coercion with `ExprSimplifier::coerce`.
-    let context = SimplifyContext::new(&props).with_schema(Arc::new(df_schema.clone()));
+    let context = SimplifyContext::default()
+        .with_schema(Arc::new(df_schema.clone()))
```

Comment:
We can probably avoid this clone via https://docs.rs/datafusion/latest/datafusion/common/struct.DFSchema.html#method.inner

So like
```suggestion
        .with_schema(Arc::clone(df_schema.inner()))
```